File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/tasks/TasksModule.kt
package au.com.crazybean.mobilex.kurir.modules.tasks

import au.com.crazybean.mobilex.kurir.extension.qualifier
import au.com.crazybean.mobilex.kurir.modules.base.ViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val tasksModule = module {
    val qualifier = qualifier<TasksWrapper>()

    // Wrapper via ViewModel
    viewModel(qualifier) {
        ViewModel(TasksWrapper(get(), get(), get()))
    }

    // Actor
    factory { (scene: TasksScene?) ->
        val viewModel = get<ViewModel<TasksWrapper>>(qualifier)
        TasksActor(scene, viewModel.wrapper)
    }
}

File: ios/Kurir/modules/tasks/TasksModule.swift
//
//  FindModule.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Mobilex

class TasksModule: Module {
    override func inject() {
        // Wrapper
        factory(TasksWrapper.self) { r in
            TasksWrapper(userData: r.resolve(UserData.self),
                         usersRepository: r.resolve(UsersRepository.self)!,
                         tasksRepository: r.resolve(TasksRepository.self)!)
        }
        
        // Actor
        factory(TasksActor.self) { (r, scene: TasksScene) -> TasksActor in
            TasksActor(scene: scene, wrapper: r.resolve(TasksWrapper.self)!)
        }
    }
}
